Documenting new default headers in HTTP probes (#24756)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

to 127.0.0.1. If your pod relies on virtual hosts, which is probably the more common
case, you should not use `host`, but rather set the `Host` header in `httpHeaders`.

> ### Default HTTP Probe Headers

Reviewer: I would set a header on line 338, 
```markdown
### HTTP probes
```
and put a paragraph here, eh
```markdown
When the kubelet sends an HTTP request as part of a probe, that request
includes headers. As well as sending headers that are mandatory in HTTP,
such as `Host:`, the kubelet sends two extra headers by default:
```

| Name |  Value | Note|
|------|----------|--|
| User-Agent | `kube-probe/` + (K8S Major version) + `.` + (K8s Minor version) e.g. `kube-probe/1.20` | |
| Accept | `*/*` | Since 1.20 |

> 

Author: Thanks, good to know. but it would not add much to this as we would still need to explain it depends on the k8s version 🤔

to 127.0.0.1. If your pod relies on virtual hosts, which is probably the more common
case, you should not use `host`, but rather set the `Host` header in `httpHeaders`.

> For an HTTP probe, the kubelet sends two additional headers in addition to standard `Host` header:

Reviewer: ```suggestion
For an HTTP probe, the kubelet sends two request headers in addition to the mandatory `Host` header:
```
?

> `User-Agent` and `Accept`. Default values for these headers are `kube-probe/1.20`
> (where `1.20` is a kubelet version) and `*/*` respectively.

Reviewer: We could definitely get the static site generator to fill in the actual version number of Kubernetes based on the version of the documentation that the reader is looking at.
```suggestion
`User-Agent` and `Accept`. The default values for these headers are `kube-probe/{{< skew latestVersion >}}`
(where `{{< skew latestVersion >}}` is the version of the kubelet ) and `*/*` respectively.
```

> When the kubelet sends an HTTP request as part of a probe, that request
> includes headers. As well as sending headers that are mandatory in HTTP,
> such as `Host`, the kubelet sends two extra headers by default: `User-Agent` and `Accept`, respectively with  `kube-probe/1.20` (where `1.20` will be your actually running Kubernetes version) and `*/*`

Reviewer 2: Suggesting simplification:

```suggestion
For an HTTP probe, the kubelet sends two additional headers in addition to standard `Host` header:
`User-Agent` and `Accept`. Default values for these headers are `kube-probe/1.20`
(where `1.20` is a kubelet version) and `*/*` respectively.
```

You can override the default headers by defining `.httpHeaders` for the probe; for example

```yaml
livenessProbe:
  httpHeaders:

>     Accept: "application/json"

Reviewer: ```suggestion
    Accept: application/json
```
?

startupProbe:
  httpHeaders:
    User-Agent: "MyUserAgent"
```

> You can also remove the headers entirely by defining them with empty values. The kubelet does not send a header if the value you set for that header is an empty string.

Reviewer 2: Is it true in general case? Or only for these two headers?

```suggestion
You can also remove these two headers by defining them with an empty value.
```
